File: 001/app/models/category.py
from enum import Enum


class CategoryType(str, Enum):
    STATIONERY = "文具类"
    ELECTRONIC = "电子耗材类"
    EQUIPMENT = "办公设备类"
    CLEANING = "清洁用品类"


class Category:
    def __init__(self, id: int, name: str, type: CategoryType, description: str = ""):
        self.id = id
        self.name = name
        self.type = type
        self.description = description

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "name": self.name,
            "type": self.type.value if hasattr(self.type, "value") else self.type,
            "description": self.description
        }

File: 088/src/models/Category.ts
import { DataTypes, Model } from 'sequelize';
import sequelize from '../config/database';
import { CategoryType } from '../types';

class Category extends Model {
  public id!: number;
  public name!: string;
  public type!: CategoryType;
  public parentId?: number;
  public sortOrder!: number;
  public isArchived!: boolean;
  public description?: string;
  public readonly createdAt!: Date;
  public readonly updatedAt!: Date;
  public readonly children?: Category[];
}

Category.init(
  {
    id: {
      type: DataTypes.INTEGER.UNSIGNED,
      autoIncrement: true,
      primaryKey: true
    },
    name: {
      type: DataTypes.STRING(100),
      allowNull: false
    },
    type: {
      type: DataTypes.ENUM(...Object.values(CategoryType)),
      allowNull: false
    },
    parentId: {
      type: DataTypes.INTEGER.UNSIGNED,
      allowNull: true,
      defaultValue: null
    },
    sortOrder: {
      type: DataTypes.INTEGER,
      defaultValue: 0
    },
    isArchived: {
      type: DataTypes.BOOLEAN,
      defaultValue: false
    },
    description: {
      type: DataTypes.TEXT
    }
  },
  {
    sequelize,
    modelName: 'Category',
    tableName: 'categories'
  }
);

Category.hasMany(Category, {
  as: 'children',
  foreignKey: 'parentId'
});

Category.belongsTo(Category, {
  as: 'parent',
  foreignKey: 'parentId'
});

export default Category;
